src/lint/lint.ts

```diff
@@ -48,37 +51,51 @@ export async function lint(options: Options, config: ProjectConfig) {
     minimumSeverity: Severity.WARNING
   });
 
-
   const analyzer = new Analyzer({
     urlLoader: new FSUrlLoader(config.root),
     urlResolver: new PackageUrlResolver(),
   });
-
   const linter = new lintLib.Linter(rules, analyzer);
 
+  // This is awkward, it would be nice to be able to get access to the analysis
+  // of a lint run rather than doing this sort of parallel construction.
+  // We want the analysis that the linter sees so that we can apply fixes
+  // to the same version of the files that the fixes were constructed for.
   let warnings;
+  let analysis;
   if (options.input) {
-    warnings = await linter.lint(options.input);
+    [warnings, analysis] = await Promise.all(
+        [linter.lint(options.input), analyzer.analyze(options.input)]);
   } else {
-    warnings = await linter.lintPackage();
+    [warnings, analysis] =
+        await Promise.all([linter.lintPackage(), analyzer.analyzePackage()]);
   }
 
   const filtered = warnings.filter((w) => !filter.shouldIgnore(w));
 
-  const printer = new WarningPrinter(
-      process.stdout, {verbosity: 'full', color: true});
-  await printer.printWarnings(filtered);
+  if (options.fix) {
+    return fix(filtered, options, config, analyzer, analysis);
```

Comment:
Rename `fixAndReport()` (or similar) for clarity, I was about to ask if we really wanted to fix silently :)

Reply:
We don't report in the same sense as `report()` does. Clarified in the jsdoc, WDYT?